unit_tests/test_redis_aggregator.py

```diff
@@ -0,0 +1,238 @@
+
+    @staticmethod
+    def test_constructor_redis_db_instantiated(aggregator):
+        assert aggregator.redis_db is not None
+
+    @staticmethod
+    def test_constructor_pubsub_instantiated(aggregator):
+        assert aggregator.pubsub is not None
+
+    @staticmethod
+    def test_constructor_client_command_buffer_instantiated(aggregator):
+        assert aggregator._client_command_buffer is not None
+
+    @staticmethod
+    def test_constructor_threadpoolexecutor_instantiated(aggregator):
+        assert aggregator.executor is not None
+
+    @staticmethod
+    def test_constructor_connect_and_subscribe_called():
+        with patch("gsy_e_sdk.redis_aggregator."
+                   "RedisAggregator._connect_and_subscribe") as mocked_method:
+            RedisAggregator(aggregator_name=TEST_AGGREGATOR_NAME)
+            mocked_method.assert_called()
+
+    @staticmethod
+    def test_constructor_lock_instantiated(aggregator):
+        assert aggregator.lock is not None
+
+
+@pytest.mark.usefixtures("mock_transaction_id_and_timeout_blocking")
+class TestRedisAggregatorDeleteAggregator:
+    """Test cases for RedisAggregator's delete_aggregator method."""
+
+    @staticmethod
+    def test_delete_aggregator_redis_db_publishes_data(aggregator):
+        data = {"name": TEST_AGGREGATOR_NAME,
+                "aggregator_uuid": TEST_AGGREGATOR_UUID,
+                "type": "DELETE",
+                "transaction_id": TEST_TRANSACTION_ID}
+
+        aggregator.aggregator_uuid = TEST_AGGREGATOR_UUID
+
+        aggregator.delete_aggregator()
+        aggregator.redis_db.publish.assert_called_with(
+            "aggregator", json.dumps(data))
+
+    @staticmethod
+    def test_delete_aggregator_api_time_out_raise_exception(aggregator):
+        with patch("gsy_e_sdk.redis_aggregator.wait_until_timeout_blocking",
+                   side_effect=AssertionError):
+            aggregator.aggregator_uuid = TEST_AGGREGATOR_UUID
+            with pytest.raises(RedisAggregatorAPIException):
+                aggregator.delete_aggregator()
+
+    @staticmethod
+    @pytest.mark.parametrize("test_is_blocking, expected_ret_val",
+                             [(True, TEST_TRANSACTION_ID), (False, None)])
+    def test_delete_aggregator_returns_expected(aggregator, test_is_blocking,
+                                                expected_ret_val):
+        aggregator.aggregator_uuid = TEST_AGGREGATOR_UUID
+        assert aggregator.delete_aggregator(is_blocking=test_is_blocking) is expected_ret_val
+
+
+@pytest.mark.usefixtures("mock_client_command_buffer_attributes")
+class TestRedisAggregatorExecuteBatchCommands:
+    """Test cases for RedisAggregator's execute_batch_commands method."""
+
+    @staticmethod
+    def test_execute_batch_commands_commands_buffer_length_zero_return_none(aggregator):
+        with patch("gsy_e_sdk.redis_aggregator.ClientCommandBuffer.buffer_length",
+                   new_callable=PropertyMock, return_value=0):
```

Comment:
Here I think we don't need to know about the `ClientCommandBuffer`. This is an implementation detail that we decided to "hide" behind the `self.commands_buffer_length` property. So we just need to know what `self.commands_buffer_length` returns.

Reply:
This, of course, only works if we also test the `commands_buffer_length` property (and it looks like you've done that).

Reply:
I'm not sure I understand what you mean here. This test was to test that when the _command_buffer_ has no elements (length zero) , the `execute_batch_commands` method returns `None` due to its own logic. 
The mock in this test is in order to fix this length zero as a return value, and therefore "force" the execute_batch_command to return `None`. 

I might not have understood what you said though.

Reply:
@ERafaelMartinez probably Pierpaolo meant that you should mock the `commands_buffer_length` property instead of mocking the `buffer_length` of the `ClientCommandBuffer`, because in fact the `execute_batch_commands` checks for the length of `commands_buffer_length`, which in its own turn checks the `buffer_length`